portality/view/admin.py

```diff
@@ -144,6 +147,83 @@ def article_endpoint(article_id):
     resp.mimetype = "application/json"
     return resp
 
+def _validate_authors(form, require=1):
+    counted = 0
+    for entry in form.authors.entries:
+        name = entry.data.get("name")
+        if name is not None and name != "":
+            counted += 1
+    return counted >= require
+
+@blueprint.route("/article/<article_id>", methods=["GET", "POST"])
+@login_required
+@ssl_required
+@write_required()
+def article_page(article_id):
+    if not current_user.has_role("edit_article"):
+        abort(401)
+    ap = models.Article.pull(article_id)
+    if ap is None:
+        abort(404)
+
+    form = ArticleForm(id=article_id)
```

Comment:
Ok, it looks like we kind of are and kind of aren't using the FormContext here, and I'm not sure what your intent is.  If we are using the FormContext, then this form should be handled within that, rather than explicitly here.  So, you would do something like:

`ArticleFormFactory.get_form_context("admin", source=ap, form_data=request.form)`

Then you would substitute "admin" for "public" if you wanted to serve the public version instead.

Reply:
Obviously you then need to write the `ArticleFormFactory` to return the appropriate FormContext instance, and plumb in the `ArticleForm` and templates in there.  I can help you with all of that.